src/edu/ucsb/nceas/metacat/startup/K8sAdminInitializer.java

```diff
@@ -0,0 +1,134 @@
+ *
+ * (In legacy deployments, these tasks are carried out manually, using the admin UI. However, this
+ * isn't possible or desirable in Kubernetes, since the site properties file is a read-only
+ * configmap)
+ */
+public class K8sAdminInitializer {
+
+    private static final Log logMetacat = LogFactory.getLog(K8sAdminInitializer.class);
+
+    /**
+     * private constructor, since this is a singleton
+     */
+    private K8sAdminInitializer() {
+    }
+
+    /**
+     * Method to be called during system startup - e.g. from the servlet <code>init()</code> method
+     * . This method, in turn, will verify that this is a containerized/k8s instance, and then
+     * will perform the necessary initialization steps
+     *
+     * @throws ServletException if anything goes wrong
+     */
+    public static void initializeK8sInstance() throws ServletException {
+
+        verifyK8s();
+
+        logMetacat.info("initializeK8sInstance() configuring for K8s environment...");
+
+        //check database and update if necessary
+        initK8sDBConfig();
+
+        // check D1 Admin settings and update if necessary
+        initK8sD1Admin();
+    }
+
+    /**
+     * Checks the environment variable "METACAT_IS_RUNNING_IN_A_CONTAINER", set by the helm chart,
+     * to verify that this instance is running in Kubernetes.
+     *
+     * @throws ServletException if this instance is NOT running in a container
+     * @implNote package private to allow for unit testing
+     */
+    static void verifyK8s() throws ServletException {
+        if (!Boolean.parseBoolean(System.getenv("METACAT_IS_RUNNING_IN_A_CONTAINER"))) {
+            throw new ServletException("initializeK8sInstance() called, but Metacat is NOT "
+                                           + "running in a container (or the "
+                                           + "METACAT_IS_RUNNING_IN_A_CONTAINER env var has not "
+                                           + "been set correctly)");
+        }
+    }
+
+    /**
+     * Call the DBAdmin code to check for and perform any database updates that may be necessary.
+     * (In a non-k8s environment, database updates are performed manually, via the admin UI)
+     *
+     * @throws ServletException if updates were unsuccessful
+     * @implNote package private to allow for unit testing
+     */
+    static void initK8sDBConfig() throws ServletException {
+
+        logMetacat.info("initK8sDBConfig(): checking for necessary database updates...");
+        MetacatVersion metacatVersion;
+        DBVersion dbVersion;
+        String mcVerStr = "NULL";
+        String dbVerStr = "NULL";
+        try {
+            metacatVersion = SystemUtil.getMetacatVersion();
+            mcVerStr = metacatVersion.getVersionString();
+            dbVersion = DBAdmin.getInstance().getDBVersion();
+            dbVerStr = dbVersion.getVersionString();
+            if (metacatVersion.compareTo(dbVersion) == 0) {
```

Comment:
Do we need to consider the scenario that the metacat version is less than the db version?

Reply:
I adopted the same logic as that used in [the DB admin page](https://github.com/NCEAS/metacat/blob/372a79e1222b70f9001048d6ab45ae4d420b5bfb/lib/admin/metacat-configuration.jsp#L123), since the DBAdmin::upgradeDatabase() method calls [DBAdmin.getUpdateScripts()](https://github.com/NCEAS/metacat/blob/d617d1480986ff89802ebe071de4fcbe88d2e504/src/edu/ucsb/nceas/metacat/admin/DBAdmin.java#L653), which iterates through the DB versions and figures out which updates need to be applied (and so it takes care of the cases where DB < metacat, **and** DB > Metacat). 

I think that's a good approach, because the logic is in one place (DBAdmin.getUpdateScripts()) instead of three places, and if we ever decide to support downgrades, we can do that simply by changing DBAdmin, without changing the calling code.

What do you think?